benchmarks/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -349,20 +351,58 @@ int main_(std::vector<std::string> const &args, const MPI_Comm comm)
     }
     }
 
-    for (int i = 0; i < n; ++i)
-      random_points_host(i) = {
-          {offset_x + random(), offset_y + random(), offset_z + random()}};
+    // Generate random points uniformly distributed within a box.
+    // FIXME: should be fixed to 1D/2D
+    auto const a = std::cbrt(n_values);
+
+    // The boxes in which the points are placed have side length two, centered
+    // around offset_[xyz] and scaled by a.
+    Kokkos::View<ArborX::Point *, DeviceType> random_points(
+        Kokkos::ViewAllocateWithoutInitializing("points"),
+        std::max(n_values, n_queries));
+    auto random_points_host = Kokkos::create_mirror_view(random_points);
+    for (int i = 0; i < random_points.extent_int(0); ++i)
+      random_points_host(i) = {{a * (offset_x + random()),
+                                a * (offset_y + random()),
+                                a * (offset_z + random())}};
     Kokkos::deep_copy(random_points, random_points_host);
+
+    Kokkos::deep_copy(
+        random_values,
+        Kokkos::subview(random_points, Kokkos::pair<int, int>(0, n_values)));
+
+    if (!shift_queries)
+    {
+      // By default, random points are "reused" between building the tree and
+      // performing queries.
+      Kokkos::deep_copy(
+          random_queries,
+          Kokkos::subview(random_points, Kokkos::pair<int, int>(0, n_queries)));
+    }
+    else
+    {
+      // For the queries, we shrink the global box by a factor three, and
+      // move it by a third of the global size towards the global center.
+      auto random_queries_host = Kokkos::create_mirror_view(random_queries);
+
+      int const max_offset = 2 * shift * i_max;
+      for (int i = 0; i < n_queries; ++i)
+        random_queries_host(i) = {
+            {a * ((offset_x + random()) / 3 + max_offset / 3),
+             a * ((offset_y + random()) / 3 + max_offset / 3),
+             a * ((offset_z + random()) / 3 + max_offset / 3)}};
```

Comment:
I just noticed that this may not be correct. Lets say you have 5 ranks. Your `i_max = std::ceil(std::cbrt(5)) = 2`. The points in `random_queries` would be correctly placed in the `x and `y` directions. However, their `z` placement would be wrong. For ranks 0-3 `offset_z` would be 0, however `max_offset` would be 4 (for `shift = 1.0`). Thus, I think the queries would stick outside of the values domain, thus resulting in no intersections for a bunch of queries.

Reply:
I mean it's just a different test case than the previous one. What else would you want the behavior to be?
Also, we should still find neighbors. It's just the radius search that might not result in intersections, isn't?

Reply:
My understanding was that we squeeze down and put queries in the middle of existing values domain. The code was not that.

>  Also, we should still find neighbors. It's just the radius search that might not result in intersections, isn't?

Yes, of course.

Reply:
> My understanding was that we squeeze down and put queries in the middle of existing values domain. The code was not that.

Yes, I agree that that's the idea and it should work for a sufficiently large of MPI processes in that way. I just don't know what else we want to do if shifting moves queries outside of the values domain. For me, adding a comment that that might happen would be good enough.
What else would you want to do in the case you are describing?

Reply:
Yeah, the case is weird, and I'm not certain about the right treatment. Lets say that we are not going to address it.